### PR TITLE
RN-321 Update Settings and Notifications UX/UI Screens

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Update the UX/UI for the Settings, Advance Settings and Notification Screens in Android and iOS

The [next ticket](https://mattermost.atlassian.net/browse/RN-321) I'm going to work on is going to be based on this PR

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-321

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on: 
* IOS 10.3 iPhone 6s and iPhone 6 simulator
* Android 6.0.1 Samsung J5
* Android 7.1.1 Nexus 5 emulator

#### Screenshots
IOS
ios

Android
android

### CODE REVIEW

File: app/screens/settings/notification_settings_mentions/notification_settings_mentions.android.js [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+                                        id: 'mobile.account_notifications.threads_mentions',
+                                        defaultMessage: 'Mentions in threads'
+                                    })}
+                                    value='never'
+                                    checked={this.state.comments === 'never'}
+                                />
+                            </RadioButtonGroup>
+                        </View>
+                        <View style={style.modalFooter}>
+                            <View style={style.divider}/>
```
Comment (enahum) [inline app/screens/settings/notification_settings_mentions/notification_settings_mentions.android.js]: Change the style to `separator`

File: app/components/settings_item/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            destructor.color = theme.errorTextColor;
+        }
+
+        let divider;
+        if (separator) {
+            divider = (<View style={style.divider}/>);
+        }
+
+        let icon;
+        if (iconName && iconType) {
```
Comment (jarredwitt) [inline app/components/settings_item/index.js]: Could we move this to a class method that returns the icon component? Just seems a bit lengthy to be in `render`

File: app/screens/settings/notification_settings/index.js [hunk 3]
```diff
@@ -9,14 +9,14 @@ import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
...
-import AccountNotifications from './account_notifications';
+import NotificationSettings from './notification_settings';
 
 function mapStateToProps(state, ownProps) {
     return {
         ...ownProps,
         config: state.entities.general.config,
-        myPreferences: state.entities.preferences.myPreferences,
         currentUser: getCurrentUser(state),
+        myPreferences: state.entities.preferences.myPreferences,
```
Comment (jarredwitt) [inline app/screens/settings/notification_settings/index.js]: Is there a selector for this that we could use?

File: app/screens/settings/notification_settings_email/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import {getTheme} from 'app/selectors/preferences';
+
+import NotificationSettingsEmail from './notification_settings_email';
+
+function mapStateToProps(state, ownProps) {
+    return {
+        ...ownProps,
+        config: state.entities.general.config,
+        myPreferences: state.entities.preferences.myPreferences,
```
Comment (jarredwitt) [inline app/screens/settings/notification_settings_email/index.js]: Here too.

File: app/screens/settings/notification_settings_mobile/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import {connect} from 'react-redux';
+
+import {getTheme} from 'app/selectors/preferences';
+
+import NotificationSettingsMobile from './notification_settings_mobile';
+
+function mapStateToProps(state, ownProps) {
+    return {
+        ...ownProps,
+        config: state.entities.general.config,
```
Comment (jarredwitt) [inline app/screens/settings/notification_settings_mobile/index.js]: Use the selector from the redux lib?

File: app/screens/settings/settings_item/index.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+        let icon;
+        if (iconName && iconType) {
+            switch (iconType) {
+            case 'fontawesome':
+                icon = (
+                    <FontAwesomeIcon
+                        name={iconName}
+                        style={[style.icon, destructor]}
+
```
Comment (jarredwitt) [inline app/screens/settings/settings_item/index.js]: Unneeded extra line right?

**Comment by enahum** [comment]
@jarredwitt changes made

File: app/screens/settings/settings_item/index.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                    {icon &&
+                    <View style={style.iconContainer}>
+                        {icon}
+                    </View>
+                    }
+                    <View style={style.wrapper}>
+                        <View style={style.labelContainer}>
+                            <FormattedText
+                                id={i18nId}
+                                defaultMessage={defaultMessage}
```
Comment (hmhealey) [inline app/screens/settings/settings_item/index.js]: Not something that needs changing, but do we really need to pass in the defaultMessage if it's coming from props? I think it still falls back to English even without that

File: app/screens/settings/notification_settings_mentions_keywords/notification_settings_mentions_keywords.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        this.props.navigator.pop();
+    };
+
+    keywordsRef = (ref) => {
+        this.keywordsInput = ref;
+    };
+
+    onKeywordsChangeText = (keywords) => {
+        if (keywords.endsWith('\n')) {
+            return this.handleSubmit();
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/settings/notification_settings_mentions_keywords/notification_settings_mentions_keywords.js]: OnSubmitEditing has some issues on Android suposed to be fixed in 49

File: app/screens/settings/settings_item/index.js (hunk 7 above)
Comment (enahum) [reply app/screens/settings/settings_item/index.js]: Just a precaution if the en.json file has it missing

**Comment by enahum** [comment]
@jarredwitt @hmhealey please take another look to this PR, I really want to merge this so I can submit the other one

**Pull request merged**